conda_smithy/configure_feedstock.py

```diff
@@ -102,10 +105,10 @@ def render_circle(jinja_env, forge_config, forge_dir):
 
 @contextmanager
 def fudge_subdir(subdir):
-    orig = conda_build.metadata.cc.subdir
-    conda_build.metadata.cc.subdir = subdir
+    orig = cc.subdir
+    cc.subdir = subdir
     yield
-    conda_build.metadata.cc.subdir = orig
+    cc.subdir = orig
```

Comment:
I have not been able to get this to work.  Even if I change the cc to directly refer to the one that that the metadata.py file used to refer to, I am still seeing the docker sh file deleted:

```
@contextmanager
def fudge_subdir(subdir):
    import conda.config as cc
    orig = cc.subdir
    cc.subdir = subdir
    yield
    cc.subdir = orig
```

I think this really needs a proper fix in conda (which I would propose as being able to set a platform argument for the relevant functions, rather than depending on platform-specific global state).  I don't know if @kalefranz would consider this a new feature (in which case, it probably would go into the 4.3 release) or whether he'd be OK putting it into older releases.